mcs/class/corlib/System.Reflection.Emit/ModuleBuilder.cs

```diff
@@ -107,18 +109,35 @@ public class ModuleBuilder : Module, _ModuleBuilder {
 
 			if (emitSymbolInfo) {
 				Assembly asm = Assembly.LoadWithPartialName ("Mono.CompilerServices.SymbolWriter");
-				if (asm == null)
-					throw new TypeLoadException ("The assembly for default symbol writer cannot be loaded");
+				if (asm == null) {
+					WarnAboutSymbolWriter("Failed to load the default Mono.CompilerServices.SymbolWriter assembly");
+					return;
+				}
 
 				Type t = asm.GetType ("Mono.CompilerServices.SymbolWriter.SymbolWriterImpl", true);
-				symbolWriter = (ISymbolWriter) Activator.CreateInstance (t, new object[] { this });
+				try {
+					symbolWriter = (ISymbolWriter) Activator.CreateInstance (t, new object[] { this });
+				} catch (System.MissingMethodException) {
+					WarnAboutSymbolWriter("The default Mono.CompilerServices.SymbolWriter is not available on this platform");					
+					return;
+				}
+				
 				string fileName = fqname;
 				if (assemblyb.AssemblyDir != null)
 					fileName = Path.Combine (assemblyb.AssemblyDir, fileName);
 				symbolWriter.Initialize (IntPtr.Zero, fileName, true);
 			}
 		}
 
+		private static void WarnAboutSymbolWriter (string message) 
+		{
+			if (has_warned_about_symbolWriter)
+				return;
+
+			has_warned_about_symbolWriter = true;
+			Console.Error.WriteLine("WARNING: {0}", message);
```

Comment:
I'm aware of that, what I was trying to say is that I think I've seen us not following that, but maybe we're just inconsistent :)